src/User.php

```diff
@@ -1173,6 +1176,7 @@ public function applyRightRules()
                             ) {
                                 unset($retrieved_dynamic_profiles[$keyretr]);
                                 unset($dynamic_profiles[$keydb]);
+                                $found = true;
```

Comment:
`$found` variable should be removed completely.
Indeed, as long as a profile is both in existing dynamic profiles (`$dynamic_profiles`) and in new computed (`$retrieved_dynamic_profiles`), value should be unset from both:
 - removed from `$dynamic_profiles` to be sure it will not be removed from DB (we want to preserve it);
 - removed from `$retrieved_dynamic_profiles` to be sure it will not be added to DB (it will be duplicated if we preserve it).

I think your code works as right it added (based on `$retrieved_dynamic_profiles` entry), so it is duplicated, then it is removed (based on `$dynamic_profiles` entry), so it is instantly deduplicated.